cache/metadata.go

```diff
@@ -29,454 +32,330 @@ const keySnapshot = "cache.snapshot"
 const keyBlobOnly = "cache.blobonly"
 const keyMediaType = "cache.mediatype"
 const keyImageRefs = "cache.imageRefs"
+const keyDeleted = "cache.deleted"
+const keyBlobSize = "cache.blobsize" // the packed blob size as specified in the oci descriptor
 
-// BlobSize is the packed blob size as specified in the oci descriptor
-const keyBlobSize = "cache.blobsize"
+// Indexes
+const blobchainIndex = "blobchainid:"
+const chainIndex = "chainid:"
 
-const keyDeleted = "cache.deleted"
+type MetadataStore interface {
+	Search(context.Context, string) ([]RefMetadata, error)
+}
 
-func queueDiffID(si *metadata.StorageItem, str string) error {
-	if str == "" {
-		return nil
-	}
-	v, err := metadata.NewValue(str)
-	if err != nil {
-		return errors.Wrap(err, "failed to create diffID value")
-	}
-	si.Update(func(b *bolt.Bucket) error {
-		return si.SetValue(b, keyDiffID, v)
-	})
-	return nil
+type RefMetadata interface {
+	ID() string
+
+	GetDescription() string
+	SetDescription(string) error
+
+	GetCreatedAt() time.Time
+	SetCreatedAt(time.Time) error
+
+	HasCachePolicyDefault() bool
+	SetCachePolicyDefault() error
+	HasCachePolicyRetain() bool
+	SetCachePolicyRetain() error
+
+	GetLayerType() string
+	SetLayerType(string) error
+
+	GetRecordType() client.UsageRecordType
+	SetRecordType(client.UsageRecordType) error
+
+	GetEqualMutable() (RefMetadata, bool)
+
+	// generic getters/setters for external packages
+	GetString(string) string
+	GetExternal(string) ([]byte, error)
+
+	SetValue(string, interface{}, string) error
```

Comment:
Where is this used where it requires to be public and `interface{}`? Current interface is weird as there is `GetString/SetValue` that don't have a matching getter/setter.

Reply:
Agreed it didn't make sense anymore now that the methods were public, changed to `SetString`